streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -471,6 +477,53 @@ public void shouldResumeConsumptionOfInitializedPartitions() {
         EasyMock.verify(consumer);
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldUpdateTasksFromPartitionAssignment() {
```

Comment:
Could we break this test up into multiple tests that are testing one thing. There is way too much going on here and it is pretty hard to understand exactly what it is testing